src/lib.rs

```diff
@@ -399,7 +395,23 @@ pub fn builder(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
                         }
                     }
 
+                    let mut error_message = vec![];
+                    // this feature uses `#[rustc_on_unimplemented]` which is only available
+                    // in a nightly compiler.
+                    if cfg!(feature = "better_error_message") {
+                        let message = format!("missing `{}`", &field_name);
+                        let label =
+                            format!("provide `{}` before calling `.build()`", &field_name);
+                        error_message.push(quote! {
+                            #[rustc_on_unimplemented(
+                                message=#message,
+                                label=#label,
+                            )]
+                        });
+                    }
+
                     constraint_traits.push(quote! {
+                        #(#error_message)*
                         trait #trait_name {}
```

Comment:
Shouldn't we have a single message per `constraint_trait`? Why `error_message` is a `vec`?